job-scraper/job_scraper.py

```diff
@@ -0,0 +1,65 @@
+import requests
+import collections
+
+
+JobOffer = collections.namedtuple(
+    "JobOffer", ("position", "company", "url", "salary", "source"),
+)
+
+
+class NoFluffJobsScraper:
+    url = "https://nofluffjobs.com/api/search/posting"
+    data_class = JobOffer
+
+    def get_offers(self):
+        return requests.get(self.url).json()["postings"]
+
+    def parse_offer(self, offer):
+        data = {
+            "position": offer.get("title"),
+            "company": offer.get("name"),
+            "url": offer.get("url"),
+            "source": "nofluffjobs",
+            "salary": "-",
+        }
+
+        return self.data_class(**data)
+
+
+class JustJoinItScraper:
+    url = "https://justjoin.it/api/offers"
+    data_class = JobOffer
+
+    def get_offers(self):
+        return requests.get(self.url).json()
+
+    def parse_offer(self, offer):
+        data = {"position": offer.get("title"),
+                "company": offer.get("company_name"),
+                "salary": "{} - {}".format(offer.get("salary_from"), offer.get("salary_to")),
+                "url": "-",
+                "source": "justjoinit",
+                }
+
+        return self.data_class(**data)
+
+
+class Scraper:
+    scraper_classes = (NoFluffJobsScraper, JustJoinItScraper)
+
+    def __init__(self):
+        self.offers = []
+
+    def run(self, filter_position=None):
+        for scraper_class in self.scraper_classes:
+            scraper = scraper_class()
+            for offer_data in scraper.get_offers():
+                offer = scraper.parse_offer(offer_data)
+                if filter_position and filter_position in offer.position.lower():
+                    self.offers.append(offer)
+        return self.offers
+
+
+scraper = Scraper()
```

Comment:
Te dwie linijki chciałbym żebyś przerzuciła do skryptu `main.py` i tam zrobiła wywołanie takie jak miałaś w poprzednim pliku `main.py` generalnie chodzi tu o pewien porządek, ponieważ jakbyś chciała zaimportować jakąś klasę z tego pliku tzn. `from job_scraper import Scraper` spowodowało by to wywołanie tych linijek i niepotrzebne ściągnięcie danych. Potraktujmy ten moduł jak "małą bilbiotekę" i nie wywołujmy tego tutaj.

Tu masz np. objaśnienie tego problemu:
https://stackoverflow.com/questions/419163/what-does-if-name-main-do

TL;DR zrób plik `main.py` w nim `if __name__ == "main" ` w którym wywołujesz np.
```
if __name__ == "__main__":
    scraper = Scraper()
    python_offers = scraper.run(filter_position="python")
    for offer in python_offers:
        print(f"{offer.position},{offer.url},{offer.salary}")
```